app/clients/db/redis_client.py

```
@@ -1,2 +1,36 @@
+class RedisClient:
+    def __init__(self, db: int = 0):
+        self.client = redis.Redis(host=settings.REDIS_ENDPOINT, port=settings.REDIS_PORT, db=db)
+
+
+    def set(self, key: str, value: str, ex: int | None = None) -> bool:
+        """Set key to value with optional expiration in seconds."""
+        try:
+            return self.client.set(key, value, ex=ex)
+        except redis.RedisError as e:
+            logger.error(f"Failed to set key {key}: {str(e)}")
+            raise RuntimeError(f"Failed to set key {key}: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)